### PR TITLE
Upgrade to v1.2.0

### PR DESCRIPTION
**Pull request by riversyang**
## All Changes

* Rename function `verify_and_apply_appchain_messages` to `verify_and_stage_appchain_messages`.
* Remove function `try_complete_switching_era` and `try_complete_distributing_reward`, add function `process_appchain_messages`.
* Add view functions for querying appchain messages and processing results of them.
* Optimize implementation and Fix bugs.

### CODE REVIEW

File: appchain-anchor/src/lookup_array.rs [hunk 1]
```diff
@@ -118,18 +118,31 @@ where
         self.end_index = *index;
     }
     ///
-    pub fn clear(&mut self) {
-        for index in self.start_index..self.end_index + 1 {
+    pub fn clear(&mut self) -> MultiTxsOperationProcessingResult {
```
Comment (hsxyl) [inline appchain-anchor/src/lookup_array.rs:121]: I have same suggestions as `appchain_messages's clear`.

**Review by hsxyl — COMMENTED** [review]
My questions focus on `fn clear`.

File: appchain-anchor/src/appchain_messages.rs [hunk 2]
```diff
@@ -130,13 +118,33 @@ impl AppchainMessages {
...
+            "Nonce range of appchain messsages: {} - {}",
+            self.min_nonce,
+            self.max_nonce
+        );
+        let mut nonce = self.min_nonce + 1;
+        while nonce <= self.max_nonce + 1 && env::used_gas() < GAS_CAP_FOR_MULTI_TXS_PROCESSING {
+            self.remove_messages_before(&nonce);
+            nonce += 1;
+        }
+        if env::used_gas() > GAS_CAP_FOR_MULTI_TXS_PROCESSING {
```
[the remark this answers is not in the document]
Comment (riversyang) [reply appchain-anchor/src/appchain_messages.rs:132]: Yes, you are right. It's better to compare `max_nonce`.

File: appchain-anchor/src/appchain_messages.rs [hunk 3]
```diff
@@ -130,13 +118,33 @@ impl AppchainMessages {
...
+            self.min_nonce = nonce - 1;
+            MultiTxsOperationProcessingResult::NeedMoreGas
+        } else {
+            self.min_nonce = 0;
+            self.max_nonce = 0;
+            MultiTxsOperationProcessingResult::Ok
+        }
```
[the remark this answers is not in the document]
Comment (riversyang) [reply appchain-anchor/src/appchain_messages.rs:139]: Will do.

File: appchain-anchor/src/lookup_array.rs (hunk 1 above)
Comment (riversyang) [reply appchain-anchor/src/lookup_array.rs:121]: Will change to compare with `end_index`.

**Review by hsxyl — APPROVED** [review]

**Pull request merged**